src/nat/data_models/span.py

```
@@ -128,10 +128,48 @@ class SpanStatus(BaseModel):
+            raise Exception("A null trace_id is invalid")
+        return v
+
+    @field_validator("span_id", mode="before")
+    @classmethod
+    def _validate_span_id(cls, v: int | str | None) -> int:
+        """Regenerate if span_id is None; raise an exception if span_id is invalid;"""
+        if isinstance(v, str):
+            try:
+                v = int(v, base=0)
+            except ValueError:
+                raise Exception("span_id is unable to be parsed")
```

Ruff (0.13.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)